### PR TITLE
[keymgr] Clarify and update the key redundancy controls

### LINKED ISSUE #16233: [keymgr] Clarify countermeasure CTRL.REDUN
Please clarify the outcome when data_en or data_valid is flipped.
As discussed, we may need to split the data_en for SW update and HW sideloads.

### PR DESCRIPTION
**Pull request by tjaychen**
- fixes #16233

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Comment by tjaychen** [comment]
@weicaiyang let me know if the documentation is clear.

File: hw/ip/keymgr/data/keymgr.hjson [hunk 1]
```diff
@@ -281,8 +281,18 @@
...
+
+        In order for a key to be populated into the software register, both the software valid
+        and the software data enable must be asserted.  The same is true for sideload.
+
+        This makes it more difficult for an attack to fault a sideload key into the software key slot.
+        An attacker would need to fault both the software valid and the software data enable.
+
+        During a sideload operation, if an attacker manages to fault the valid but not the data enable,
+        the software key is populated with random data.  If an atacker manages to fault the data enable but
+        not the valid, then the software key retains its previous value.
```
Comment (weicaiyang) [inline hw/ip/keymgr/data/keymgr.hjson:295]: I pan to  test it as follows. LMK if this sequence doesn't sound right to you.

1. start a sideload operation
2. flip either data_sw_en or data_valid, 
3. read sw_share* and check it doesn't match the sideload output

**Review by weicaiyang — APPROVED** [review]
LGTM with a question, thanks

File: hw/ip/keymgr/data/keymgr.hjson (hunk 1 above)
Comment (tjaychen) [reply hw/ip/keymgr/data/keymgr.hjson:295]: yep! I think for 3, you can go one step further..
if valid is flipped but data_en is not, you should see data that doesn't match either the previously flopped value or the sideload value.

if valid is not flipped but data_en is, you should see the previous value.

Comment (weicaiyang) [reply hw/ip/keymgr/data/keymgr.hjson:295]: sounds good. Thanks Tim

**Pull request merged**